akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala

```diff
@@ -447,42 +484,330 @@ class InterpreterSpec extends InterpreterSpecKit {
 
     }
 
-    "handle failure inside op" in new TestSetup(Seq(Map((x: Int) ⇒ if (x == 0) throw TE else x))) {
+    "emit failure when op throws" in new TestSetup(Seq(Map((x: Int) ⇒ if (x == 0) throw TE else x, stoppingDecider))) {
       lastEvents() should be(Set.empty)
 
       downstream.requestOne()
       lastEvents() should be(Set(RequestOne))
+      upstream.onNext(2)
+      lastEvents() should be(Set(OnNext(2)))
+
+      downstream.requestOne()
+      lastEvents() should be(Set(RequestOne))
+      upstream.onNext(0) // boom
+      lastEvents() should be(Set(Cancel, OnError(TE)))
+    }
 
+    "resume when op throws" in new TestSetup(Seq(Map((x: Int) ⇒ if (x == 0) throw TE else x, resumingDecider))) {
+      lastEvents() should be(Set.empty)
+
+      downstream.requestOne()
+      lastEvents() should be(Set(RequestOne))
       upstream.onNext(2)
       lastEvents() should be(Set(OnNext(2)))
 
       downstream.requestOne()
       lastEvents() should be(Set(RequestOne))
+      upstream.onNext(0) // boom
+      lastEvents() should be(Set(RequestOne))
 
-      upstream.onNext(0)
+      downstream.requestOne()
+      lastEvents() should be(Set(RequestOne))
+      upstream.onNext(3)
+      lastEvents() should be(Set(OnNext(3)))
+    }
+
+    "emit failure when op throws in middle of the chain" in new TestSetup(Seq(
+      Map((x: Int) ⇒ x + 1, stoppingDecider),
+      Map((x: Int) ⇒ if (x == 0) throw TE else x + 10, stoppingDecider),
+      Map((x: Int) ⇒ x + 100, stoppingDecider))) {
+
+      lastEvents() should be(Set.empty)
+
+      downstream.requestOne()
+      lastEvents() should be(Set(RequestOne))
+      upstream.onNext(2)
+      lastEvents() should be(Set(OnNext(113)))
+
+      downstream.requestOne()
+      lastEvents() should be(Set(RequestOne))
+      upstream.onNext(-1) // boom
       lastEvents() should be(Set(Cancel, OnError(TE)))
+    }
 
+    "resume when op throws in middle of the chain" in new TestSetup(Seq(
+      Map((x: Int) ⇒ x + 1, resumingDecider),
+      Map((x: Int) ⇒ if (x == 0) throw TE else x + 10, resumingDecider),
+      Map((x: Int) ⇒ x + 100, resumingDecider))) {
+
+      lastEvents() should be(Set.empty)
```

Comment:
I don't think you need to repeat this line in all test cases. It is relevant for the basic interpreter spec, but I think these supervision tests can rely on that normal operation "works as expected". Your call though.